consensus/core/src/block_validator.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::block::Block;
+use async_trait::async_trait;
+
+/// The interfaces to validate the legitimacy of a statement block's contents.
+#[async_trait]
+pub trait BlockValidator: Send + Sync + 'static {
```

Comment:
I also like the `BlockVerifier` name from the mysticeti codebase because we have the validator component for starting up the consensus node. Unless you want to change the component from validator -> node.

Reply:
Ok I'll stick to the `BlockVerifier`